sdk/src/framework/sc/shared/PlayerScore.kt

```diff
@@ -0,0 +1,52 @@
+package sc.shared
+
+import com.thoughtworks.xstream.annotations.XStreamAlias
+import com.thoughtworks.xstream.annotations.XStreamAsAttribute
+import com.thoughtworks.xstream.annotations.XStreamImplicit
+import java.lang.IllegalArgumentException
+import java.math.BigDecimal
+
+@XStreamAlias(value = "score")
+data class PlayerScore(
+        @XStreamAsAttribute
+        val cause: ScoreCause?,
+        @XStreamAsAttribute
+        val reason: String,
+        @XStreamImplicit(itemFieldName = "part")
+        val parts: Array<BigDecimal>
+) {
+    
+    constructor(winner: Boolean, reason: String): this(ScoreCause.REGULAR, reason, if (winner) 2 else 0)
+    constructor(cause: ScoreCause?, reason: String, vararg scores: Int): this(cause, reason, scores.map { BigDecimal(it) }.toTypedArray())
+    
+    fun size(): Int = parts.size
+    
+    val values: List<BigDecimal>
```

Comment:
Didn't we want to get rid of BigDecimal?
Or is there a good reason to keep it?

Reply:
I don't think the scores will ever go out of int range. Can't think of a good reason.

Reply:
If we average them, they may not be int anymore. And if we get floating-point errors in a championship scoring, we have a problem. Better safe than sorry I'd say.